internal/pkg/docker/dockerfile/dockerfile.go

```
@@ -89,31 +114,62 @@ func (df *Dockerfile) parse() error {
 	file, err := df.fs.Open(df.path)
 
 	if err != nil {
-		return fmt.Errorf("read dockerfile: %w", err)
+		return fmt.Errorf("read Dockerfile: %w", err)
 	}
 	defer file.Close()
 
-	scanner := bufio.NewScanner(file)
-	parsedDockerfile := parseFromScanner(scanner)
+	f, err := afero.ReadFile(df.fs, file.Name())
+	if err != nil {
+		return fmt.Errorf("read Dockerfile %s error: %w", f, err)
+	}
+	fileString := string(f)
+
+	parsedDockerfile, err := parseFromReader(fileString)
+	if err != nil {
+		return fmt.Errorf("parse Dockerfile: %w", err)
```

```suggestion
		return err
```

Our [guideline](https://www.efekarakus.com/golang/2019/09/26/when-to-wrap-errors.html#recommendation-) is to wrap any errors coming from public functions (our dependencies), but since `parseFromReader` is private we can just pass the error as is.